### ISSUE TITLE
Ability to specify timeout time?

### ISSUE THREAD

**Issue by BatuhanK** [paypal/PayPal-node-SDK#134]
Hello, can we specify timeout (in ms) at requests to PayPal?

**Comment by bluk** [comment]
There is no generic timeout method available in this SDK right now. Just thinking right now but would you like a global timeout or timeouts for specific requests? Also, for timeouts, there are 2 different ideas where the timeout is the total amount of time allowed for the request (so even if you are receiving a partial response, the request times out) or it only times out if there is no response at all. For integrations, I assume the latter is more favorable but want to confirm.

**Comment by BatuhanK** [comment]
Right now we are facing performance issues at payment.create() method especially when request made to the ip below(after ~15-20 second it gives timeout).
`[Error: connect ETIMEDOUT 173.0.84.66:443]
  code: 'ETIMEDOUT',
  errno: 'ETIMEDOUT',
  syscall: 'connect',
  address: '173.0.84.66',
  port: 443 }`

I'm just asking is there any settings for timeouts? If not, it would be nice to have an optional parameter for setting request timeouts for api access.

**Comment by bluk** [comment]
There are no options to set the request timeout today in the SDK.

I did a few tests, and I don't think setting a [`request.setTimeout`](https://nodejs.org/api/http.html#http_request_settimeout_timeout_callback) will help since that method (and the socket setTimeout method) takes effect only after the connection is made. Going by StackOverflow and other general posts, the timeout set is only for timeout on inactivity on a socket after the socket has been connected. `ETIMEDOUT` indicates the connection was not made at all.

It appears to be an OS level control for connection timeouts (googling for `ETIMEDOUT`). You may want to look at how to change it at the OS level and see if that's safe to change for your environment.
